lib/util/utils.ts

```diff
@@ -229,6 +229,20 @@ const XML2JS_PARSER_OPTS: xml2js.OptionsV2 = {
   explicitRoot: false
 };
 
+// TODO: remove xml2js, use synchronous parsing, remove dispatchRequest()
+export function parseXML(xmlString: string): Promise<any> {
+  const xmlParser = new xml2js.Parser(XML2JS_PARSER_OPTS);
+  return new Promise(function(resolve: (result: any) => void, reject: (err: any) => void) {
+    xmlParser.parseString(xmlString, function(err: any, result: any) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(result);
+      }
+    });
+  });
+}
```

Comment:
Can this function just be async instead of creating and returning a Promise?

Reply:
This is promisifying a callback-only API. Sometime soon, I hope it will simply be synchronous